cinder/volume/drivers/vmware/vmdk.py

```diff
@@ -493,6 +494,20 @@ def _get_volume_stats(self):
                         'storage_profile': datastore["storage_profile"],
                         'connection_capabilities': connection_capabilities,
                         }
+
+                # Add any custom attributes associated with the datastore
+                ds_object = datastore.get("datastore_object", None)
+                custom_attributes = {}
+                if ds_object:
+                    # see if there are any custom values
+                    attrs = self.volumeops.get_datastore_custom_attributes(
+                        ds_object
+                    )
```

Comment:
Doing one request to the vCenter per datastore looks quite inefficient. We are able to construct requests filtering for managed-objects and could by that query the whole bunch of datastores and their attributes at once. One example in Nova would be [this](https://github.com/sapcc/nova/blob/stable/rocky-m3/nova/virt/vmwareapi/vm_util.py#L1676-L1683).

Reply:
The code on the branch changed and I didn't reference a specific commit at the time, so the link now goes somewhere else. I think I was referring to this: https://github.com/sapcc/nova/blob/84f3e06855a1f83678ef09c841802f64cf613b17/nova/virt/vmwareapi/vm_util.py#L1618-L1629